### ISSUE TITLE
proto.py文件中forward方法的疑问

### ISSUE THREAD

**Issue by lushishuai** [thunlp/FewRel#30]
你好，我在跑5-way-5-shot Few-Shot Relation Classification  model: proto  encoder: cnn，在训练的时候trainN参数为10。
在proto.py 文件代码中：
 support = torch.mean(support, 2) # Calculate prototype for each class
        logits = -self.__batch_dist__(support, query) # (B, total_Q, N)
        minn, _ = logits.min(-1)
        logits = torch.cat([logits, minn.unsqueeze(2) - 1], 2) # (B, total_Q, N + 1)
        _, pred = torch.max(logits.view(-1, N+1), 1)
        return logits, pred
logits 与minn进行拼接形成新的logits，维度变为N+1=11。我们设置了trainN为10，这样求交叉熵的时候，拼接的minn部分并没有发挥任何用处。我想问一下作者这个minn有什么其他我没有想到的用处么？期待您的回答！十分感谢！

**Comment by gaotianyu1350** [comment]
因为本身proto就不能很好的支持NOTA这个设定，这里拼接一维只是权宜之计，让proto能勉强做这个任务，并没有办法直接优化
